freqtrade/freqai/data_kitchen.py

```diff
@@ -0,0 +1,1318 @@
+import copy
+import datetime
+import json
+import logging
+import pickle as pk
```

Comment:
What's the reason / benefit of using pickle in some cases, and  `joblib.dump`/load  in different cases (within the same file)? 
i'd assume that joblib (and it's implicit cloudpickle library for pickle) is the preferred way - (mainly since we do have that available /installed alread)

Reply:
cloudpickle is vendored in joblib - check hyperopt.py on how to import it ... 
i actually think that the `dump` and `load` functions you import are either direct re-exports from cloudpickle - or wrappers for it.

Reply:
Yup, it all worked. Just needed to change the protocol to `cloudpickle.DEFAULT_PROTOCOL`